glide-core/src/iam/mod.rs

```diff
@@ -0,0 +1,1063 @@
+use aws_config::BehaviorVersion;
+use aws_credential_types::{Credentials, provider::ProvideCredentials};
+use aws_sigv4::http_request::{
+    SignableBody, SignableRequest, SignatureLocation, SigningSettings, sign,
+};
+use aws_sigv4::sign::v4;
+use logger_core::{log_error, log_info, log_warn};
+use std::sync::Arc;
+use std::time::Duration;
+use std::time::SystemTime;
+use thiserror::Error;
+use tokio::sync::{Notify, RwLock};
+use tokio::task::JoinHandle;
+use tokio::time::{MissedTickBehavior, interval};
+
+/// Maximum refresh interval in seconds (12 hours)
+const MAX_REFRESH_INTERVAL_SECONDS: u32 = 12 * 60 * 60; // 43200 seconds
+/// Default refresh interval in seconds (14 minutes)
+const DEFAULT_REFRESH_INTERVAL_SECONDS: u32 = 14 * 60; // 840 seconds
+/// Warning threshold for refresh interval in seconds (15 minutes)
+/// Setting refresh intervals above this value may have performance consequences
+const WARNING_REFRESH_INTERVAL_SECONDS: u32 = 15 * 60; // 900 seconds
+/// SigV4 presign expiration (15 minutes)
+const TOKEN_TTL_SECONDS: u64 = 15 * 60; // 900
+
+/// Custom error type for IAM operations in Glide
+#[derive(Debug, Error)]
+pub enum GlideIAMError {
+    #[error(
+        "IAM authentication error: Invalid refresh interval. Must be between 0 and {max}, got: {actual}"
+    )]
+    InvalidRefreshInterval { max: u32, actual: u32 },
+
+    #[error("IAM authentication error: Failed to get AWS credentials: {0}")]
+    CredentialsError(String),
+
+    #[error("IAM authentication error: Token generation failed: {0}")]
+    TokenGenerationError(String),
+
+    #[error("IAM authentication error: {0}")]
+    Other(String),
+}
+
+/// Service type configuration for IAM authentication
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum ServiceType {
+    ElastiCache,
+    MemoryDB,
+}
+
+impl ServiceType {
+    fn service_name(&self) -> &'static str {
+        match self {
+            ServiceType::ElastiCache => "elasticache",
+            ServiceType::MemoryDB => "memorydb",
+        }
+    }
+}
```

Comment:
```suggestion
#[derive(Copy, Clone, Debug, PartialEq, Eq, AsRefStr, Display, EnumString)]
pub enum ServiceType {
    #[strum(serialize = "elasticache")] // Matches previous service_name output
    ElastiCache,
    #[strum(serialize = "memorydb")]   // Matches previous service_name output
    MemoryDB,
}
```
We already using strum macro in redis-rs so no new dependency